### PR TITLE
50145_6_7_admin_book_RUD

### PR DESCRIPTION
**Pull request by manazirahsan**
This pull request contains admin/books' index, show, edit and delete. BookImages will be added after this pull request.
This pull request contains task #50145, #50146, #50147

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,7 +34,8 @@ gem "sdoc", "~> 0.4.0", group: :doc
 # gem "capistrano-rails", group: :development
 
 gem "nested_form"
-
+gem "will_paginate"
```
Comment (abdullahallmehedi) [inline Gemfile]: use kaminari

File: app/views/admin/books/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, t(:books_page_title)) %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/edit.html.erb]: ()

File: app/views/admin/books/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<% provide(:title, t(:books_page_title)) %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: ()

File: app/views/admin/books/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<% provide(:title, t(:books_page_title)) %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: ()

File: app/views/admin/books/_form.html.erb [hunk 5]
```diff
@@ -20,10 +20,11 @@
       <div class="form-group">
         <%= f.label t(:book_form_book_category), class: "control-label" %>
         <%= f.collection_select :category_id, Category.all, :id, :name,
-          prompt: "--Select A Category--", class: "form-control" %>
+          prompt: t(:book_form_prompt_for_category), class: "form-control" %>
       </div>
-      <div class="col-md-2 col-md-offset-5">
+      <div class="col-md-4 col-md-offset-4">
         <%= f.submit t(:book_form_submit_button), class: "btn btn-primary" %>
+        <%= link_to t(:book_form_cancel_button), admin_books_path, class: "btn btn-default" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/_form.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+<div class="panel panel-default">
+  <div class="panel-heading">
+    <div class="panel-title">
+      <div class="clearfix">
+        <%= t :books_page_title %>
+        <div class="pull-right">
+          <%= link_to t(:book_index_create_btn), new_admin_book_path, class: "btn btn-primary" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+              <%= book.publish_date %>
+            </td>
+            <td>
+              <%= book.num_of_pages %>
+            </td>
+            <td>
+              <%= book.category.name %>
+            </td>
+            <td>
+              <%= link_to t(:book_index_edit_btn), edit_admin_book_path(book), class: "btn btn-xs btn-warning" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+              <%= link_to t(:book_index_delete_btn), admin_book_path(book), method: :delete, data: {confirm: t(:make_sure)}, class: "btn btn-xs btn-danger" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+              <%= link_to t(:book_index_review_btn), "#", class: "btn btn-xs btn-info" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+              <%= link_to t(:book_index_favorite_btn), "#", class: "btn btn-xs btn-success" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+              <%= link_to t(:book_index_status_read_btn), "#", class: "btn btn-xs btn-default" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/index.html.erb]: 80

File: app/views/admin/books/new.html.erb [hunk 12]
```diff
@@ -1,3 +1,3 @@
-<% provide(:title, "Book title" ) %>
+<% provide(:title, t(:books_page_title)) %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/new.html.erb]: ()

File: app/views/admin/books/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+<div class="panel panel-default">
+  <div class="panel-heading">
+    <div class="panel-title">
+      <div class="clearfix">
+        <%= link_to t(:books_show_all_books_btn), admin_books_path, class: "btn btn-primary" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: 80

File: app/views/admin/books/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+        <div class="pull-right">
+          <%= link_to t(:book_index_edit_btn), edit_admin_book_path(@book), class: "btn btn-md btn-warning" %>
+          <%= link_to t(:book_index_delete_btn), admin_book_path(@book), method: :delete, data: {confirm: t(:make_sure)}, class: "btn btn-md btn-danger" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: 80

File: app/views/admin/books/show.html.erb [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+          <%= link_to t(:book_index_review_btn), "#", class: "btn btn-md btn-info" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: 80

File: app/views/admin/books/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+          <%= link_to t(:book_index_favorite_btn), "#", class: "btn btn-md btn-success" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: 80

File: app/views/admin/books/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+          <%= link_to t(:book_index_status_read_btn), "#", class: "btn btn-md btn-default" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/show.html.erb]: 80

File: db/seeds.rb [hunk 18]
```diff
@@ -4,4 +4,17 @@
 # Examples:
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
\ No newline at end of file
+#   Mayor.create(name: 'Emanuel', city: cities.first)
```
Comment (abdullahallmehedi) [inline db/seeds.rb]: remove all unnecessary lines

File: Gemfile (hunk 1 above)
Comment (manazirahsan) [inline Gemfile]: I see, kaminari is better alternative for will_paginate.
https://github.com/amatsuda/kaminari
I may adapt it next time :)
I am taking care other convention error.

File: app/views/admin/books/edit.html.erb (hunk 2 above)
Comment (manazirahsan) [inline app/views/admin/books/edit.html.erb]: If I remove the braces then it show a syntax error.
I am taking care of other convention error.

File: app/views/admin/books/index.html.erb (hunk 3 above)
Comment (manazirahsan) [inline app/views/admin/books/index.html.erb]: Like above.

File: app/views/admin/books/show.html.erb (hunk 4 above)
Comment (manazirahsan) [inline app/views/admin/books/show.html.erb]: Like above.

File: app/views/admin/books/_form.html.erb (hunk 5 above)
Comment (manazirahsan) [inline app/views/admin/books/_form.html.erb]: Exactly.
I am always ignorance about such convention.
I need to amend myself!
Thanks for your cordial review :)

File: app/views/admin/books/new.html.erb (hunk 12 above)
Comment (manazirahsan) [inline app/views/admin/books/new.html.erb]: Does not work without braces.

File: db/seeds.rb (hunk 18 above)
Comment (manazirahsan) [inline db/seeds.rb]: Removed.

File: app/views/admin/books/edit.html.erb (hunk 2 above)
Comment (MuhammadTamzid) [inline app/views/admin/books/edit.html.erb]: <% provide :title, t(:books_page_title) %>
You can use this

**Pull request closed without merging**